sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/EquivalentExpressions.scala

```diff
@@ -140,6 +140,7 @@ class EquivalentExpressions {
     case i: If => i.predicate :: Nil
     case c: CaseWhen => c.children.head :: Nil
     case c: Coalesce => c.children.head :: Nil
+    case n: NaNvl => n.left :: Nil
```

Comment:
Can you update the comment of `childrenToRecurse` method too?

Reply:
sure, add the comment